toolchain/lex/tokenized_buffer.cpp

```diff
@@ -313,6 +311,26 @@ auto TokenizedBuffer::PrintToken(llvm::raw_ostream& output_stream,
   output_stream << " },";
 }
 
+auto TokenizedBuffer::FindLineIndexImpl(int32_t offset) const -> LineIndex {
+  CARBON_DCHECK(!line_infos_.empty());
+  const auto* line_it =
+      std::partition_point(line_infos_.begin(), line_infos_.end(),
+                           [offset](const LineInfo& line_info) {
```

Comment:
Is the reference here deliberate/required? This is a small struct, so IIUC it's more efficient if we can not have it:

```suggestion
                           [offset](LineInfo line_info) {
```